### PR TITLE
CI: cancel previous runs for pull requests

### PR DESCRIPTION
**Pull request by chenxsan**

### CODE REVIEW

File: .github/workflows/cancel.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Cancel
+on:
+  workflow_run:
+    workflows: ["Ensure PR"]
+    types:
+      - requested
```
Comment (snitin315) [inline .github/workflows/cancel.yml:6]:
```suggestion
on:
  pull_request:
```

File: .github/workflows/cancel.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+jobs:
+  cancel:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: styfle/cancel-workflow-action
+      with:
+        workflow_id: ${{ github.event.workflow.id }}
```
Comment (snitin315) [inline .github/workflows/cancel.yml:13]:
```suggestion
        workflow_id: testing.yml
        access_token: ${{ github.token }}
```

**Review by snitin315 — COMMENTED** [review]
You can run on PR event

File: .github/workflows/cancel.yml (hunk 1 above)
Comment (chenxsan) [reply .github/workflows/cancel.yml:6]: Nah, it won't run against those pull requests from forks https://github.com/styfle/cancel-workflow-action#advanced-pull-requests-from-forks. E.g, I'm submitting this pull request from my forked repository.

Comment (snitin315) [reply .github/workflows/cancel.yml:6]: [Try this](https://github.com/webpack-contrib/css-loader/blob/master/.github/workflows/cancel.yml#L2), we have similar in webpack-contrib and it runs on PR form forks.

Comment (chenxsan) [reply .github/workflows/cancel.yml:6]: Thanks. Any reason to avoid `workflow_run`? Seems fine to me according to the documentation.

Comment (chenxsan) [reply .github/workflows/cancel.yml:6]: Maybe we can only test it after getting it merged, not in this one.

Comment (snitin315) [reply .github/workflows/cancel.yml:6]: I'm good with it if it works 👍

**Pull request merged**